src/Ladybug_SunPath.py

```diff
@@ -509,8 +509,9 @@ def main(latitude, longitude, timeZone, elevation, north, hour, day, month, time
                     listInfo[i][5] = (1,1,1)
                     listInfo[i][6] = (12,31,24)
                     #find the numbers
-                    for h, hr in enumerate(sunUpHours):
-                        value = selList[int(math.floor(hr))] + (selList[int(math.ceil(hr))] - selList[int(math.floor(hr))])* (hr - math.floor(hr))
+                    for h, hr in enumerate(sunUpHours):                     
+#                        value = selList[int(math.floor(hr))] + (selList[int(math.ceil(hr))] - selList[int(math.floor(hr))])* (hr - math.floor(hr))
+                        value = selList[int(hr-1)]

```

Comment:
@mostaphaRoudsari ,
This was exactly my concern and I actually checked it already.  It seems the rest of the code already deals with this well by using the same epw hourly value for the sub-hourly timestep.  So @devngc 's  change is fine in this respect.
The only thing that I want to check now before merging is whether this causes the component to fail at times when there is sun at midnight (in the arctic or antarctic).  @devngc , if you could check that and make sure that your fix works here, I'll feel comfortable merging this in.